build-data-capturing-gradle-samples/capture-quality-check-issues/gradle-quality-check-issues.gradle.kts

```diff
@@ -0,0 +1,129 @@
+import com.gradle.enterprise.gradleplugin.GradleEnterpriseExtension
+import com.gradle.scan.plugin.BuildScanExtension
+import groovy.xml.XmlSlurper
+import groovy.xml.slurpersupport.NodeChildren
+import groovy.xml.slurpersupport.NodeChild
+import groovy.xml.slurpersupport.GPathResult
+
+/**
+ * This Gradle script captures issues found by reporting tasks,
+ * and adds these as custom values.
+ */
+
+project.extensions.configure<GradleEnterpriseExtension>() {
+    buildScan {
+        gradle.taskGraph.beforeTask {
+            if (ReportingTask.isSupported(this) && this is Reporting<*>) {
+                val reportTask = this as Reporting<ReportContainer<*>>
+                reportTask.reports.withGroovyBuilder { "xml" { setProperty("enabled", true) } }
```

Comment:
oh wow, this get's quite gnarly with kotlin since we don't have any of the generated accessors here.

Reply:
Yeah it's not pretty, let me know if you find a better way